### PR TITLE
feat: support use of Context with RequestBuilder

### LINKED ISSUE #77: Support for context timeout/deadline for requests in underlying HTTP client
**Is your feature request related to a problem? Please describe.**
The default behavior for whenever a request is made is using `http.NewRequest` [(from here)](https://github.com/IBM/go-sdk-core/blob/master/v4/core/request_builder.go#L309) which will use the default HTTP clients timeout [(from here)](https://github.com/IBM/go-sdk-core/blob/master/v4/core/base_service.go#L86). This will block the caller and force them to wait until the request either fulfills or times out.

**Describe the solution you'd like**
Support passing in a context throughout different request types, e.g.:
```go
ctx, cancelFunc := context.WithDeadline(context.Background(), someTime)
client.GetDocumentAsStream(ctx, s.NewGetDocumentOptions(someDatabaseName, someID))
```
By supporting this, the request can be canceled so that the caller isn't blocked by the library if they have their own timeout they need to support.

### PR DESCRIPTION
**Pull request by padamstx**
Fixes arf/planning-sdk-squad#2230
Fixes #77

This commit adds support for setting a context.Context
instance on the RequestBuilder struct.  This Context instance
will then be associated with the http.Request instance constructed
by the RequestBuilder.Build() method.

BREAKING CHANGE: Minimum supported Go version is now 1.13.
The minimum version of Go supported by this library has been changed
to Go version 1.13. To use this and future versions of the
Go core library, please make sure that you are using
Go version 1.13 or higher.

### CODE REVIEW

File: v4/core/request_builder.go [hunk 1]
```diff
@@ -318,18 +328,23 @@ func (requestBuilder *RequestBuilder) Build() (*http.Request, error) {
...
+			return
 		}
 		requestBuilder.Body = newBody
 		requestBuilder.Header.Add(CONTENT_ENCODING, "gzip")
 	}
 
 	// Create the request
-	req, err := http.NewRequest(requestBuilder.Method, requestBuilder.URL.String(), requestBuilder.Body)
+	if requestBuilder.RequestContext != nil {
+		req, err = http.NewRequestWithContext(*requestBuilder.RequestContext, requestBuilder.Method, requestBuilder.URL.String(), requestBuilder.Body)
```
Comment (jorge-ibm) [inline v4/core/request_builder.go:342]: Should the pointer on `*requestBuilder.RequestContex` be removed? The build is failing here.

Comment (padamstx) [reply v4/core/request_builder.go:342]: Nope, that build failure was due to the fact that http.NewRequestWithContext method was added in Go 1.13, while we're trying to build with Go 1.12.

File: .bumpversion.cfg [hunk 2]
```diff
@@ -3,16 +3,10 @@ current_version = 4.6.1
 commit = True
 message = Update version numbers from {current_version} -> {new_version} [skip ci]
 
-[bumpversion:file:v4/core/version.go]
+[bumpversion:file:v5/core/version.go]
 search = __VERSION__ = "{current_version}"
 replace = __VERSION__ = "{new_version}"
 
-[bumpversion:file:v4/go.mod]
```
Comment (padamstx) [inline .bumpversion.cfg:10]: We really don't need to update the go.mod file with bumpversion.

File: .travis.yml [hunk 3]
```diff
@@ -3,23 +3,19 @@ language: go
...
 go:
-- 1.12.x
+- 1.13.x
 
 notifications:
   email: false
 
-env:
-  global:
-    - GO111MODULE=on
```
Comment (padamstx) [inline .travis.yml:13]: No longer need this env var.

File: .travis.yml [hunk 4]
```diff
@@ -3,23 +3,19 @@ language: go
...
-
 before_install:
   - nvm install 12
   - npm install -g npm@6.x
   - sudo apt-get update
   - sudo apt-get install python
 
 install:
-  - curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.21.0
+  - curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(go env GOPATH)/bin v1.31.0
```
Comment (padamstx) [inline .travis.yml:18]: Might as well spruce up our toolset while we're creating a new major version :)

**Comment by padamstx** [comment]
This PR has been updated to use Go v 1.13 and will result in a new major version of the Go SDK.

**Review by jorge-ibm — APPROVED** [review]
LGTM

**Comment by padamstx** [comment]
Closing this PR because in favor of an alternate solution that does not require a new major version of the Go core.

**Pull request closed without merging**